### PR TITLE
core: warn if people use the undocumented/depreacted ConditionNull=

### PR DESCRIPTION
**Pull request by poettering**
Triggered by:

https://github.com/systemd/systemd/issues/11812

### CODE REVIEW

File: src/core/load-fragment.c [hunk 1]
```diff
@@ -2512,6 +2512,8 @@ int config_parse_unit_condition_null(
         assert(rvalue);
         assert(data);
 
+        log_syntax(unit, LOG_WARNING, filename, line, 0, "ConditionNull= is deprecated, please do not use.");
```
Comment (yuwata) [inline src/core/load-fragment.c]: Please use `%s` with `lvalue` instead of `ConditionNULL=`, as this is also used for `AssertNull=`.

**Comment by poettering** [comment]
Pushed a new version, with the change @yuwata asked for. ptal!

File: src/core/load-fragment.c (hunk 1 above)
Comment (poettering) [reply src/core/load-fragment.c]: indeed!

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**